src/visualizers/VisualizerChaos.ts

```diff
@@ -0,0 +1,393 @@
+import p5 from "p5";
+import { VisualizerDefault } from './VisualizerDefault';
+import { VisualizerParamsSchema, VisualizerExportModule } from './VisualizerInterface';
+import { ParamType } from '@/shared/ParamType';
+import { ValidationStatus } from '@/shared/ValidationStatus';
+
+// p5 Colour palette class
+class Palette {
+	colorList: p5.Color[] = [];
+	backgroundColor: p5.Color;
+	textColor: p5.Color;
+
+	constructor( sketch: p5, hexList?: string[], hexBack?: string, hexText?: string ) {
+		let myHexList: string[] = [];
+		let myHexBack = '#000000';
+		let myHexText = '#000000';
+		if ( hexList !== undefined ) { myHexList = hexList; }
+		if ( hexBack !== undefined ) { myHexBack = hexBack; }
+		if ( hexText !== undefined ) { myHexText = hexText; }
```

Comment:
That's cool syntax, except it didn't work for me since it didn't change the type of the variables (they are still `string | undefined` instead of `string`, for example) causing typescript to choke.  Not sure if I did it wrong?  Instead I'm putting default values in the function parameters, and that works fine.

Reply:
Weird. `??` should act as a type guard against undefined. Can't be certain what's up without seeing the exact code and error message, but if you have an equivalently straightforward syntax working, there's probably no point in taking the time to convey all that detail

Reply:
I probably did something wrong.  But the alternative solution seemed nicer anyway?

Reply:
Yes, with the default arguments it is very nice and compact.